### PR TITLE
Upgrade to guava 18.0

### PR DESCRIPTION
**Pull request by aleph-zero**

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataConnectionCreator.java [hunk 1]
```diff
@@ -66,8 +67,8 @@ public DataTunnel getTunnel(DrillbitEndpoint endpoint) {
     return new DataTunnel(newManager);
   }
 
-  public void close() {
-    Closeables.closeQuietly(server);
+  public void close() throws IOException {
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataConnectionCreator.java]: No need for `throws IOException`, `Closeables.close(..., true)` makes sure it's not propagated

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/UserServer.java [hunk 2]
```diff
@@ -279,7 +279,10 @@ public ProtobufLengthDecoder getDecoder(BufferAllocator allocator, OutOfMemoryHa
 
   @Override
   public void close() throws IOException {
-    Closeables.closeQuietly(authenticator);
-    super.close();
+    try {
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/UserServer.java]: why are you wrapping this inside a `try` clause ? do you suspect authenticator.close() can throw other exceptions other than `IOException` ?

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataClientConnection.java [hunk 3]
```diff
@@ -84,8 +85,8 @@ public boolean equals(Object obj) {
     return true;
   }
 
-  public void shutdownIfClient() {
-    Closeables.closeQuietly(client);
+  public void shutdownIfClient() throws IOException {
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataClientConnection.java]: No need for throws IOException, Closeables.close(..., true) makes sure it's not propagated

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControllerImpl.java [hunk 4]
```diff
@@ -60,11 +62,13 @@ public ControlTunnel getTunnel(DrillbitEndpoint endpoint) {
...
 
-  public void close() {
-    Closeables.closeQuietly(server);
-    for (ControlConnectionManager bt : connectionRegistry) {
-      bt.close();
+  public void close() throws IOException {
+    try {
+      Closeables.close(server, true);
+    } finally {
+      for (ControlConnectionManager bt : connectionRegistry) {
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControllerImpl.java]: If you are trying to make sure all resources are closed even if `server.close()` throws a runtime exception, then you are only solving half the problem. if `bt.close()` throws an exception then the loop may not proceed through the connection registry

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControlConnection.java [hunk 5]
```diff
@@ -110,7 +110,9 @@ public boolean equals(Object obj) {
 
   public void shutdownIfClient() {
     if (bus.isClient()) {
-      Closeables.closeQuietly(bus);
+      try {
+        Closeables.close(bus, true);
+      } catch (Exception e) { /* do nothing */ }
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControlConnection.java]: Any particular reason you added this ? is it safe to just ignore runtime exceptions here ?

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/ImplCreator.java [hunk 6]
```diff
@@ -74,7 +74,7 @@ public static RootExec getExec(FragmentContext context, FragmentRoot root) throw
       root = IteratorValidatorInjector.rewritePlanWithIteratorValidator(context, root);
     }
     final ImplCreator creator = new ImplCreator();
-    Stopwatch watch = new Stopwatch();
+    Stopwatch watch = Stopwatch.createStarted();
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/ImplCreator.java]: you forgot to remove `watch.start()`

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/sort/SortTemplate.java [hunk 7]
```diff
@@ -47,7 +47,7 @@ public void setup(FragmentContext context, SelectionVector4 vector4, VectorConta
 
   @Override
   public void sort(SelectionVector4 vector4, VectorContainer container){
-    Stopwatch watch = new Stopwatch();
+    Stopwatch watch = Stopwatch.createStarted();
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/sort/SortTemplate.java]: forgot to remove `watch.start()`

File: exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/SpoolingRawBatchBuffer.java [hunk 8]
```diff
@@ -376,7 +376,7 @@ public long getBodySize() {
     }
 
     public void writeToStream(FSDataOutputStream stream) throws IOException {
-      Stopwatch watch = new Stopwatch();
+      Stopwatch watch = Stopwatch.createStarted();
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/SpoolingRawBatchBuffer.java]: you forgot to remove watch.start()

File: exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/SpoolingRawBatchBuffer.java [hunk 9]
```diff
@@ -423,7 +423,7 @@ public void readFromStream() throws IOException, InterruptedException {
           final long check = stream.readLong();
           pos = stream.getPos();
           assert check == this.check : String.format("Check values don't match: %d %d, Position %d", this.check, check, currentPos);
-          Stopwatch watch = new Stopwatch();
+          Stopwatch watch = Stopwatch.createStarted();
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/SpoolingRawBatchBuffer.java]: same here

**Comment by adeneche** [comment]
can you please create a JIRA for this as well ?

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataConnectionCreator.java (hunk 1 above)
Comment (aleph-zero) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataConnectionCreator.java]: Yes, ```Closeables.close(..., true)``` does indeed swallow any IOException. However, the ```Closeables.close()``` method still declares that it throws IOException and so we have to deal with that somehow. 

So what would you recommend? We can either add a redundant catch block for IOException that we know will never be executed, or we can modify the ```DataConnectionManager.close()``` to declare it to throw IOE.

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/UserServer.java (hunk 2 above)
Comment (aleph-zero) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/UserServer.java]: You are right, I think leaving out the ```try``` is better.

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataClientConnection.java (hunk 3 above)
Comment (aleph-zero) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataClientConnection.java]: Same comment as above, re: either need to declare or catch IOException, despite the fact that ```Closeables.close()``` may never actually throw IOE.

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControllerImpl.java (hunk 4 above)
Comment (aleph-zero) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControllerImpl.java]: Addressed

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControlConnection.java (hunk 5 above)
Comment (aleph-zero) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ControlConnection.java]: So how about if we change to ```public void shutdownIfClient() throws IOException``` and remove the ```catch```?

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/sort/SortTemplate.java (hunk 7 above)
Comment (aleph-zero) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/sort/SortTemplate.java]: Fixed

File: exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/SpoolingRawBatchBuffer.java (hunk 8 above)
Comment (aleph-zero) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/SpoolingRawBatchBuffer.java]: Fixed

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/ImplCreator.java (hunk 6 above)
Comment (aleph-zero) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/ImplCreator.java]: Fixed

File: exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/SpoolingRawBatchBuffer.java (hunk 9 above)
Comment (aleph-zero) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/SpoolingRawBatchBuffer.java]: Fixed

File: exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataConnectionCreator.java (hunk 1 above)
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/rpc/data/DataConnectionCreator.java]: You can use `AutoCloseables.close(server, logger)` which catch any exception and log it as a warning in the passed logger

**Pull request closed without merging**